nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,524 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.sympa;
+  pkg = pkgs.sympa.override { inherit (cfg) dataDir; };
+  dataDir = cfg.dataDir;
+  fqdns = mapAttrsToList (name: _val: name) cfg.domains;
+  usingNginx = (cfg.web.enable && cfg.web.server == "nginx");
+
+  sympaSubServices = [
+    "sympa-archive.service"
+    "sympa-bounce.service"
+    "sympa-bulk.service"
+    "sympa-task.service"
+  ];
+
+  sympaServiceConfig = srv: {
+    Type = "forking";
+    Restart = "always";
+    ExecStart = "${pkg}/bin/${srv}.pl";
+    PIDFile = "/run/sympa/${srv}.pid";
+    User = cfg.user;
+    Group = cfg.group;
+    RuntimeDirectory = "sympa";
+    RuntimeDirectoryPreserve = true;
+  };
+
+  configVal = value:
+    if value == true then "on"
+    else if value == false then "off"
+    else toString value;
+  configGenerator = c: concatStrings (flip mapAttrsToList c (key: val: "${key}\t${configVal val}\n"));
+
+  mainConfig = pkgs.writeText "sympa.conf" (configGenerator cfg.settings);
+  robotConfig = fqdn: domain: pkgs.writeText "${fqdn}-robot.conf" (configGenerator domain.settings);
+
+  transport = pkgs.writeText "transport.sympa" (concatStringsSep "\n" (flip map fqdns (domain: ''
+    ${domain}                        error:User unknown in recipient table
+    sympa@${domain}                  sympa:sympa@${domain}
+    listmaster@${domain}             sympa:listmaster@${domain}
+    bounce@${domain}                 sympabounce:sympa@${domain}
+    abuse-feedback-report@${domain}  sympabounce:sympa@${domain}
+  '')));
+
+  virtual = pkgs.writeText "virtual.sympa" (concatStringsSep "\n" (flip map fqdns (domain: ''
+    sympa-request@${domain}  postmaster@localhost
+    sympa-owner@${domain}    postmaster@localhost
+  '')));
+
+  listAliases = pkgs.writeText "list_aliases.tt2" ''
+    #--- [% list.name %]@[% list.domain %]: list transport map created at [% date %]
+    [% list.name %]@[% list.domain %] sympa:[% list.name %]@[% list.domain %]
+    [% list.name %]-request@[% list.domain %] sympa:[% list.name %]-request@[% list.domain %]
+    [% list.name %]-editor@[% list.domain %] sympa:[% list.name %]-editor@[% list.domain %]
+    #[% list.name %]-subscribe@[% list.domain %] sympa:[% list.name %]-subscribe@[%list.domain %]
+    [% list.name %]-unsubscribe@[% list.domain %] sympa:[% list.name %]-unsubscribe@[% list.domain %]
+    [% list.name %][% return_path_suffix %]@[% list.domain %] sympabounce:[% list.name %]@[% list.domain %]
+  '';
+in
+{
+
+  ###### interface
+  options.services.sympa = with types; {
+
+    enable = mkEnableOption "Sympa mailing list manager";
+
+    dataDir = mkOption {
+      type = path;
+      default = "/var/lib/sympa";
```

Comment:
Removed. I wonder if it's worth it to keep `user` and `group` configurable?

Reply:
It might be. What is the use case you were thinking of?

Reply:
I've got no use case for configurable user/group, added it to the module because many other modules have it, not sure whether it's really useful though.

Reply:
In that case I would suggest it is not worth keeping.